### DISCUSSION TITLE
ESP32/TTL-RS485 Protective Earth connection?

### DISCUSSION THREAD

**Discussion by Goaheadz** [omerfaruk-aran/esphome_samsung_hvac_bus#51] [General]
Hi Everyone,

Just a quick question regarding Protective earth.

I see no mention of protective earth in the web, regarding RS485 when using ESP and TTL/RS485 converters.
Since HVACs comunication exists outdoor/indoor and its susceptible to lightning strikes, shouldnt we connect protective earth? (question mark in red in picture bellow)

I am not an HVAC expert, so I dont know if this is already done in all indoor and outdoor units.

![image](https://github.com/lanwin/esphome_samsung_ac/assets/13796554/c8084d9b-9f74-4426-9179-390c811d4a35)

Advise needed.

Regards
Goahead

**Comment by Goaheadz** [comment]
And yes...I already had fried HVAC boards in outdoor unit.
Over AC voltage or Ligthing strikes, dont know, but microprocessor died.

So lets play safe, and try to understand if earth is needed in that convertion board.

**Comment by Goaheadz** [comment]
Someone design the schematic from the above TTL-RS485 board.
My 2 cents, I think that if we want ligthing protection,(we should, since we are talking about outdoor and indoor machines connected to AC lines) those 2 blue underline GND, should be connected to protection earth. not digital ground.
That way, TVS would be able to give some protection. 

![image](https://github.com/lanwin/esphome_samsung_ac/assets/13796554/43b8b91b-8a8f-4879-9da8-0865750f968e)

Share a thought.

**Comment by lanwin** [comment]
Sadly I have no clue since my hardware knowledge is pretty limited. Maybe one of the users have a clue on that CC @mbo18 @bzumik1 @Foxhill67

**Comment by mbo18** [comment]
No clue here. I’m using an M5Stack atom which is connected using 12V/GND directly from the HVAC. So exactly like any Samsung wired remote

**Comment by Goaheadz** [comment]
After going a bit more dip into the subject, the correct answer is "It depends".

RS485 as a common-mode range of -7 to 12V that will be rejected by the receiver. (noise subtraction in differential pair at receiver )
When nodes draw power from different sources (different sections of electrical installation and distance) this range can be exceeded.
In the case of range exceeded, ground should be connected to allow better reference point. from transmitter to receiver (same 0).

As we notice in most HVAC installations, F1 and F2 are routed using normal electrical cable, and star topology is used instead of fly-by.
Some filtering is used in outdoor and indoor circuits to filter noise and keep common mode range within spec.
Usually there is capacitors connecting 485 pair to protective ground.

People that use M5Stack is already using common digital ground due to DC power supply being provided by HVAC. 
Don't know the M5Stack circuit.
 
Others that use simpler rs485 circuits, may not use DC from HVAC (DC-DC), but AC to DC converters.
AC to DC usually are isolated and they dont even use protective ground connection.

So it really depends.
I can measure common mode before hand, but I will connect RS485 without any ground to TVSs.
1- Connection to RS485 and AC will be very short.
2 - Second test, I will get 12V DC from HVAC as it will be safer to connect and convert.

I will let you know as soon as I have news (waiting for aliexpress mail :))
